app/components/frame.js

```diff
@@ -1,8 +1,20 @@
 import Component from '@ember/component';
-import { get, set, computed } from '@ember/object';
-import { sort, notEmpty, equal } from '@ember/object/computed';
-import { inject as service } from '@ember/service';
+import { action, set } from '@ember/object';
+export default class FrameComponent extends Component {
+    tagName = ''
+    sqrtPixels = null;
 
-export default Component.extend({
-  
-});
+    init() {
+      super.init(...arguments)
+      this.sqrtPixels = Math.sqrt(this.pixels.length);
+    }
+
+    @action
+    updatePixelColor(pixelIndex) {
+        const newColor = this.paletteColor;
+        const newPixels = [...this.pixels];
+        newPixels[pixelIndex] = newColor;
+        set(this, 'pixels', newPixels);
+    }
+}
```

Comment:
Now that we're doing glimmer stuff — I'd love to see how you'd want to model this component using `@tracked` (I think this is actually a really interesting use-case!)